e3sm_diags/driver/lat_lon_driver.py

```diff
@@ -177,22 +156,30 @@ def run_diag(parameter):
                             )
                         )
 
-                        # Regrid towards the lower resolution of the two
-                        # variables for calculating the difference.
-                        mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
-                            mv1_domain,
-                            mv2_domain,
-                            parameter.regrid_tool,
-                            parameter.regrid_method,
-                        )
+                        if parameter.ref_name != "":
```

Comment:
The logic in lines 159-198 and 231-266 are the same, so I would extract them as an internal function to avoid code duplication and allow for re-usability.

Example: 

```
def _create_and_save_metrics(parameter, mv1, mv1_domain, mv2_domain):
    if parameter.ref_name != "":
        # Regrid towards the lower resolution of the two
        # variables for calculating the difference.
        mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
                            mv1_domain,
                            mv2_domain,
                            parameter.regrid_tool,
                            parameter.regrid_method,
                        )

        diff = mv1_reg - mv2_reg
        metrics_dict = create_metrics(
                            mv2_domain, mv1_domain, mv2_reg, mv1_reg, diff
                        )
    else:
        mv2_domain = None
        mv2_reg = None
        diff = None
        metrics_dict = create_metrics(
                            mv2_domain, mv1, mv2_reg, mv1, diff
                        )

    # Saving the metrics as a json.
    metrics_dict["unit"] = mv1.units
    fnm = os.path.join(
                        utils.general.get_output_dir(parameter.current_set, parameter),
                        parameter.output_file + ".json",
                    )
    with open(fnm, "w") as outfile:
        json.dump(metrics_dict, outfile)

    # Get the filename that the user has passed in and display that.
    fnm = os.path.join(
                        utils.general.get_output_dir(parameter.current_set, parameter),
                        parameter.output_file + ".json",
                    )
    logger.info(f"Metrics saved in {fnm}")
    return mv2_domain, diff, metrics_dict
```

Then replace lines 159-198 and 231-254 with:
```
                    mv2_domain, diff, metrics_dict = _create_and_save_metrics(
                        parameter, mv1, mv1_domain, mv2_domain
                    )
```

There are better ways of writing the new `_create_and_save_metrics()` function (e.g., break that up into smaller functions), but I'll leave that up to you. It would make unit testing easier in the future if tests aren't written in this PR.